sdk/identity/azure-identity/setup.py

```diff
@@ -72,7 +72,7 @@
     ),
     python_requires=">=3.7",
     install_requires=[
-        "azure-core<2.0.0,>=1.11.0",
+        "azure-core<2.0.0,>=1.23.1",
```

Comment:
Let's not force users to upgrade azure-core. We can use try import catch instead?

Reply:
This was motivated by the TokenCredential import, but that may not actually be necessary. I talked this over with Laurent, and he feels that the best solution would be to actually match the method signatures of the protocol; i.e. document kwargs like `claims` even if they wouldn't have an effect for a particular credential. I think this does make sense, since it makes the capabilities and limitations of each credential more explicit and more correctly implements the protocol.

I want to first check what pyright thinks of this PR's approach. If explicit subtyping without full implementation raises errors with pyright anyway, then this approach won't be viable regardless and would make the choice easier to make (unless you agree that we should go for the full implementation route regardless).